### ISSUE TITLE
Java: xml output of preformatted (`<pre>`) block adds para-block for blank lines (hindering certain manual parsing)

### ISSUE THREAD

**Issue by Stewori** [doxygen/doxygen#8127]
**Describe the bug**
Processing a Java code snippet like
```
public class prfmt_test {

    /**
     * <pre>{@literal
     * public MyManagedArray(MyManagedArray toCopy) {
     * }
     *
     * public Object clone() {
     * }
     * }</pre>
     */
    public void test() {
    }
```
results in an xml snippet
```
...
        <detaileddescription>
<para><preformatted>{@literal
public MyManagedArray(MyManagedArray toCopy) {
}</preformatted></para>
<para><preformatted>public Object clone() {
}
}</preformatted> </para>
        </detaileddescription>
...
```
Note the paragraph start on the blank line although no `<p>` is present in the doc comment.

**Expected behavior**
I expect the xml snippet to be
```
...
        <detaileddescription>
<para><preformatted>{@literal
public MyManagedArray(MyManagedArray toCopy) {
}

public Object clone() {
}
}</preformatted> </para>
        </detaileddescription>
...
```

**To Reproduce**
Find a Java file and doxygen config attached. The issue is demonstrated like mentioned above and with some variants, e.g. without `@literal` block.
After running doxygen on the example, `xml/classprfmt__test.xml` contains descriptions showing the issue explained above.

**Version**
Tested with self-compiled snapshot from 2020-10-24, 1.9.0 (5f34e8ae667c24900e61c72e7dfc213d53a7cb05).

`lsb_release -a`:
```
LSB Version:	core-9.20160110ubuntu0.2-amd64:core-9.20160110ubuntu0.2-noarch:security-9.20160110ubuntu0.2-amd64:security-9.20160110ubuntu0.2-noarch
Distributor ID:	LinuxMint
Description:	Linux Mint 18 Sarah
Release:	18
Codename:	sarah
```

**Additional context**
The issue also occurs without `@literal`, but the example with `@literal` points out a case where it is most severe.
This issue makes manual parsing of `@literal` blocks (which are not supported by doxygen as of https://github.com/doxygen/doxygen/issues/4953) impossible or very painful at best as soon as they are nested in `<pre>` blocks. This is because if you walk the xml tree the `@literal` block would close in a different branch than where it opens. During such a walk it is hard to know or guess where the block would end. Even worse: a typical use case is displaying Java code which itself makes use of the {} characters.
Nesting `@literal` in `<pre>` appears to be a common design pattern (unfortunately) https://blogs.oracle.com/darcy/javadoc-tip:-code-and-literal.

[bugreport_preformatted.zip](https://github.com/doxygen/doxygen/files/5432271/bugreport_preformatted.zip)

**Comment by albert-github** [comment]
Regarding the missing line, and the extra block:
```
</preformatted></para>
<para><preformatted>
```
this is even more evident when we don't have 1 empty line but e.g. 10, in that case it shows also only one extra block.
It is also present in e.g. the HTML output where it manifests itself as `</pre><pre>`.

Note: same problem also occurs when using cpp code.

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #8133 for the missing space problem.

**Comment by albert-github** [comment]
Code has been integrated in master on github (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
